stl/inc/atomic

```diff
@@ -115,7 +115,48 @@ _NODISCARD extern "C" bool __cdecl __std_atomic_has_cmpxchg16b() noexcept;
 #define ATOMIC_LLONG_LOCK_FREE    2
 #define ATOMIC_POINTER_LOCK_FREE  2
 
+// Padding bits should not participate in cmpxchg comparison starting in C++20.
+// Clang does not have __builtin_zero_non_value_bits to exclude these bits to implement this C++20 feature.
+// The EDG front-end substitutes everything and runs into incomplete types passed to atomic<T>.
+#if _HAS_CXX20 && !defined(__clang__) /* TRANSITION, LLVM-46685 */ && !defined(__EDG__)
+#define _CMPXCHG_MASK_OUT_PADDING_BITS 1
+#else
+#define _CMPXCHG_MASK_OUT_PADDING_BITS 0
+#endif
+
 _STD_BEGIN
+// STRUCT TEMPLATE _Storage_for
+struct _Form_mask_t {};
+_INLINE_VAR constexpr _Form_mask_t _Form_mask;
+
+template <class _Ty>
+struct _Storage_for {
+    // uninitialized space to store a _Ty
+    alignas(_Ty) unsigned char _Storage[sizeof(_Ty)];
+
+    _Storage_for()                    = default;
+    _Storage_for(const _Storage_for&) = delete;
+    _Storage_for& operator=(const _Storage_for&) = delete;
+
+    explicit _Storage_for(_Form_mask_t) noexcept {
+        _CSTD memset(_Storage, 0xff, sizeof(_Ty));
+        __builtin_zero_non_value_bits(_Ptr());

```

Comment:
The tests are failing in GitHub with:
```
"D:\build\x86\out\inc\atomic", line 143: error: expression must be a pointer to
          a complete object type
          __builtin_zero_non_value_bits(_Ptr());
                                        ^
```

Reply:
Compiler: `sizeof Ty`, totally fine? Other intrinsic? How could you, that's not complete (!?)

Reply:
Oh I see, it's because I exposed EDG to the intrinsic. It's scary that the github system and not Contest.